uaclient/security_status.py

```diff
@@ -191,3 +202,276 @@ def security_status(cfg: UAConfig) -> Dict[str, Any]:
+                    updates=updates_esm_infra,
+                    pl_updates="s" if updates_esm_infra > 1 else "",
+                    packages=packages_infra,
+                    pl_packages="s" if packages_infra > 1 else "",
+                )
+            )
+            if is_attached:
+                print("")
+                print(
+                    messages.SS_SERVICE_COMMAND.format(
+                        service="ESM Infra", command="ua enable esm-infra"
+                    )
+                )
+
+    if (
+        is_lts
+        and packages_apps
+        and esm_apps_status == ApplicationStatus.DISABLED
+    ):
+        print("")
+        print(
+            messages.SS_SERVICE_ADVERTISE.format(
+                service="Ubuntu Pro with ESM Apps",
+                updates=updates_esm_apps,
+                pl_updates="s" if updates_esm_apps > 1 else "",
+                packages=packages_apps,
+                pl_packages="s" if packages_apps > 1 else "",
+            )
+        )
+        if is_attached:
+            print("")
+            print(
+                messages.SS_SERVICE_COMMAND.format(
+                    service="ESM Apps", command="ua enable esm-apps"
+                )
+            )
+
+    if is_lts and not is_attached:
+        print("")
+        print(messages.SS_UNATTACHED)
+
+
+def security_status(cfg: UAConfig):
+    packages_summary = security_status_dict(cfg)["summary"]
+
+    esm_infra_status = ESMInfraEntitlement().application_status()[0]
+    esm_apps_status = ESMAppsEntitlement().application_status()[0]
+
+    _print_package_summary(packages_summary, esm_infra_status, esm_apps_status)
+    _print_info_messages(packages_summary)
+    _print_esm_actions(packages_summary, esm_infra_status, esm_apps_status)
+
+
+def list_third_party_packages():
+    cache = Cache()
+
+    installed_packages = [package for package in cache if package.is_installed]
+    third_party_packages = [
+        package
+        for package in installed_packages
+        if get_origin_for_package(package) == "third-party"
+    ]
+
+    total_packages = len(installed_packages)
+    width = len(str(total_packages))
+
+    print(messages.SS_SUMMARY_TOTAL.format(total_packages=total_packages))
+
+    if third_party_packages:
+        msg = messages.SS_SUMMARY_THIRD_PARTY_SN
+        if len(third_party_packages) > 1:
+            msg = messages.SS_SUMMARY_THIRD_PARTY_PL
+        print(
+            msg.format(
+                count="{:>{width}}".format(
+                    len(third_party_packages), width=width
+                )
+            )
+        )
+
+        print("")
+        print(messages.SS_THIRD_PARTY)
+
+        print("")
+        print("Packages:")
+        for package in third_party_packages:
+            print(package.name)
+
+        print("")
+        print(
+            messages.SS_POLICY_HINT.format(
+                package=choice(third_party_packages).name
+            )
+        )
```

Comment:
This bit is really similar to the `list_unavailable_packages`, maybe we can isolate it in a helper function